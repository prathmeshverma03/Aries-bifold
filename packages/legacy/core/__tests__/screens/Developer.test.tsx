import { render } from '@testing-library/react-native'
import React from 'react'

import Developer from '../../App/screens/Developer'
import { testIdWithKey } from '../../App/utils/testable'

describe('Developer screen', () => {
  beforeEach(() => {
    // Silence console.error because it will print a warning about Switch
    // "Warning: dispatchCommand was called with a ref ...".
    jest.spyOn(console, 'error').mockImplementation(() => {
      return null
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  test('screen renders correctly', () => {
    const tree = render(<Developer />)

    expect(tree).toMatchSnapshot()
  })

  test('pressables exist', async () => {
    const { findByTestId } = render(<Developer />)

    const VerifierToggle = await findByTestId(testIdWithKey('ToggleVerifierCapability'))
    const ConnectionToggle = await findByTestId(testIdWithKey('ToggleConnectionInviterCapabilitySwitch'))
    const DevVerifierToggle = await findByTestId(testIdWithKey('ToggleDevVerifierTemplatesSwitch'))

    expect(VerifierToggle).not.toBe(null)
    expect(ConnectionToggle).not.toBe(null)
    expect(DevVerifierToggle).not.toBe(null)
  })
})
