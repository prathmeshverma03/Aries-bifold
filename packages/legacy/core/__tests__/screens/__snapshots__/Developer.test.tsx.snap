// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Developer screen screen renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#42803E",
      "borderRadius": 3,
      "borderWidth": 1,
      "marginHorizontal": 20,
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "normal",
        },
        Object {
          "margin": 10,
        },
      ]
    }
  >
    Place content here you would like to make available to developers when developer mode is enabled.
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        accessible={false}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlign": "left",
          }
        }
      >
        Verifier.UseVerifierCapability
      </Text>
    </View>
    <View
      accessibilityLabel="Verifier.Toggle"
      accessibilityRole="switch"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
        }
      }
      testID="com.ariesbifold:id/ToggleVerifierCapability"
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="rgba(53, 130, 63, 0.35)"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "backgroundColor": "#D3D3D3",
              "borderRadius": 16,
            },
          ]
        }
        thumbTintColor="#606060"
        tintColor="#D3D3D3"
        value={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlign": "left",
          }
        }
      >
        Connection.UseConnectionInviterCapability
      </Text>
    </View>
    <View
      accessibilityLabel="Connection.Toggle"
      accessibilityRole="switch"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
        }
      }
      testID="com.ariesbifold:id/ToggleConnectionInviterCapabilitySwitch"
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="rgba(53, 130, 63, 0.35)"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "backgroundColor": "#D3D3D3",
              "borderRadius": 16,
            },
          ]
        }
        thumbTintColor="#606060"
        tintColor="#D3D3D3"
        value={false}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginVertical": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginRight": 10,
            "textAlign": "left",
          }
        }
      >
        Verifier.UseDevVerifierTemplates
      </Text>
    </View>
    <View
      accessibilityLabel="Verifier.ToggleDevTemplates"
      accessibilityRole="switch"
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
        }
      }
      testID="com.ariesbifold:id/ToggleDevVerifierTemplatesSwitch"
    >
      <RCTSwitch
        accessibilityRole="switch"
        onChange={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTintColor="rgba(53, 130, 63, 0.35)"
        style={
          Array [
            Object {
              "height": 31,
              "width": 51,
            },
            Object {
              "backgroundColor": "#D3D3D3",
              "borderRadius": 16,
            },
          ]
        }
        thumbTintColor="#606060"
        tintColor="#D3D3D3"
        value={false}
      />
    </View>
  </View>
</View>
`;
